modules/data.remote/R/GEDI_AGB_prep.R

```diff
@@ -56,481 +92,437 @@ GEDI_AGB_prep <- function(site_info, time_points, outdir = file.path(getwd(), "G
-      point.lat.lon <- matrix(c(mean(extent[c("ymin", "ymax")]), mean(extent[c("xmin", "xmax")])), nrow = 1) %>% 
-        `colnames<-`(c("lat", "lon")) %>%
-        as.data.frame
-      extent.x.y <- data.frame(matrix(c(extent["xmin"], extent["ymin"],
-                                         extent["xmax"], extent["ymin"],
-                                         extent["xmax"], extent["ymax"],
-                                         extent["xmin"], extent["ymax"]), nrow = 4, byrow = T)) %>% `colnames<-`(c("lon", "lat"))
-      res <- utils::read.csv(file.path(site_folder, "GEDI_AGB.csv"))
-      ggplot2::ggplot() +
-        ggplot2::geom_polygon(data = extent.x.y, ggplot2::aes(x = .data$lon, y = .data$lat), color="blue", fill = "white") +
-        ggplot2::geom_point(data = res, ggplot2::aes(x = .data$lon_lowestmode, y = .data$lat_lowestmode, color = .data$agbd)) +
-        ggplot2::geom_point(shape = 24, data = point.lat.lon, ggplot2::aes(x = .data$lon, y = .data$lat), size = 3) +
-        ggplot2::geom_text(data = point.lat.lon, ggplot2::aes(x = .data$lon, y = .data$lat, label=site.id, hjust=-0.1, vjust=0)) +
-        ggplot2::scale_color_distiller(palette = 'Greens', direction = 1) +
-        ggplot2::labs(color = "AGB")
-    }
-  }
-}
-#' Extract L4A GEDI above ground biomass data for the GEDI AGB prep function.
-#'
-#' @param site_info A list including site_id, longitude, and latitude.
-#' @param start_date target start date (date with YYYY-MM-DD format) for preparing GEDI AGB from remote or local database.
-#' @param end_date end date (date with YYYY-MM-DD format) for preparing GEDI AGB from remote or local database.
-#' @param outdir Directory where the final CSV file will be stored.
-#' @param nfile.min the minimum required file number to be downloaded and extracted, default is 0.
-#' @param nrow.min the minimum required observation number to be extracted, default is 0.
-#' @param buffer buffer distance (in degrees) for locate GEDI AGB searching box (default is 0.01 [~ 1 km]).
-#' @param gradient the gradient for iteratively enlarge the extent if the nfile.min or nrow.min are not reached, default is 0. If nfile.min or nrow.min is 0 this will be skipped.
+#' Detect which GEDI level 4A tiles intercept which site.
 #'
-#' @return A list of AGB data frames for each site.
+#' @title GEDI_L4A_Finder
+#' @param files Character: full paths of GEDI level 4A tiles.
+#' @param site_info List: list of site info including site_id, site_name, lon, and lat.
+#' @param buffer Numeric: buffer distance (in degree) that is used to create the bounding box (default is 0.005 [~ 500 m]).
+#' @param cores Numeric: numbers of core to be used for the parallel computation. The default is the maximum current CPU number.
 #' 
-#' @examples
-#' @author Dongchen Zhang
-#' @importFrom magrittr %>%
-#' @importFrom rlang .data
-GEDI_AGB_extract <- function(site_info, start_date, end_date, outdir, nfile.min = 0, nrow.min = 0, buffer = 0.01, gradient = 0) {
```

Comment:
I see that you've replaced with download and extract functions with new functions that do the download and aggregations. That's conceptually fine, but I do wonder about how these functions fit into an overall workflow, as I'm not seeing any changes to the code that is calling these functions. My recollection was that there was a naming pattern and function argument pattern to the download and extract functions that was designed to provide predictability and scalability to the overall workflow. How do these new functions fit into that? Does it make sense to retain those high-level functions and have them call these new functions?